utils/gyb_syntax_support/AttributeNodes.py

```diff
@@ -197,5 +199,122 @@
          ]),
 
     # objc-selector -> objc-selector-piece objc-selector?
-    Node('ObjCSelector', kind='SyntaxCollection', element='ObjCSelectorPiece')
+    Node('ObjCSelector', kind='SyntaxCollection', element='ObjCSelectorPiece'),
+
+    # The argument of '@differentiable(...)'.
+    # differentiable-attr-arguments ->
+    #     differentiation-params-clause? ','?
+    #     differentiable-attr-func-specifier? # jvp
+    #     differentiable-attr-func-specifier? # vjp
+    #     generic-where-clause?
+    Node('DifferentiableAttributeArguments', kind='Syntax',
+         description='''
+         The arguments for the `@differentiable` attribute: an optional \
+         differentiation parameter list and associated functions.
+         ''',
+         children=[
+             Child('DiffParams', kind='DifferentiationParamsClause',
+                   is_optional=True),
+             Child('DiffParamsComma', kind='CommaToken', description='''
+                   The comma following the differentiation parameters clause,
+                   if it exists.
+                   ''', is_optional=True),
+             Child('MaybePrimal', kind='DifferentiableAttributeFuncSpecifier',
```

Comment:
Note: `MaybePrimal` and `MaybeAdjoint` should be removed.
A corresponding change should be made to [apple/swift-syntax](https://github.com/apple/swift-syntax) to update gyb-generated sources, e.g. https://github.com/apple/swift-syntax/pull/178.